### PR TITLE
optee_rust_examples_ext: Migrate to standard environment vars

### PR DESCRIPTION
**Pull request by b49020**
Allow OP-TEE rust examples to be compiled using standard OP-TEE environment variables:
- CROSS_COMPILE
- TA_DEV_KIT_DIR
- OPTEE_CLIENT_EXPORT

### CODE REVIEW

**Comment by b49020** [comment]
@DemesneGH Fyi..

File: common.mk [hunk 1]
```diff
@@ -196,11 +196,13 @@ CROSS_COMPILE_NS_USER   ?= "$(CCACHE)$(AARCH$(COMPILE_NS_USER)_CROSS_COMPILE)"
 CROSS_COMPILE_NS_KERNEL ?= "$(CCACHE)$(AARCH$(COMPILE_NS_KERNEL)_CROSS_COMPILE)"
 CROSS_COMPILE_S_USER    ?= "$(CCACHE)$(AARCH$(COMPILE_S_USER)_CROSS_COMPILE)"
 CROSS_COMPILE_S_KERNEL  ?= "$(CCACHE)$(AARCH$(COMPILE_S_KERNEL)_CROSS_COMPILE)"
+CROSS_COMPILE_RUST	?= "$(AARCH$(COMPILE_S_KERNEL)_CROSS_COMPILE)"
```
Comment (jenswi-linaro) [inline common.mk:199]: This is user space only so `COMPILE_S_USER` would be more appropriate.

File: br-ext/package/optee_rust_examples_ext/optee_rust_examples_ext.mk [hunk 2]
```diff
@@ -28,7 +29,11 @@ export CC = $(OPTEE_DIR)/toolchains/aarch64/bin/aarch64-linux-gnu-gcc
 define OPTEE_RUST_EXAMPLES_EXT_BUILD_CMDS
 	@$(foreach f,$(wildcard $(@D)/examples/*/Makefile), \
 		echo Building $f && \
-		$(OPTEE_RUST_EXAMPLES_TC_PATH_ENV) $(MAKE) -C $(dir $f) &&) true
+		$(OPTEE_RUST_EXAMPLES_TC_PATH_ENV) \
+		CROSS_COMPILE="$(shell echo $(BR2_PACKAGE_OPTEE_RUST_EXAMPLES_EXT_CROSS_COMPILE))" \
```
Comment (jenswi-linaro) [inline br-ext/package/optee_rust_examples_ext/optee_rust_examples_ext.mk:33]: Why is the `$(shell echo ...)` part needed?

File: common.mk (hunk 1 above)
Comment (b49020) [reply common.mk:199]: Makes sense, I will use that instead.

File: br-ext/package/optee_rust_examples_ext/optee_rust_examples_ext.mk (hunk 2 above)
Comment (b49020) [reply br-ext/package/optee_rust_examples_ext/optee_rust_examples_ext.mk:33]: It was copied from `optee_examples_ext.mk` but it works without them too. So I will drop `$(shell echo ...)` part.

**Comment by b49020** [comment]
Addressed review comments.

File: br-ext/package/optee_rust_examples_ext/Config.in [hunk 3]
```diff
@@ -11,6 +11,12 @@ config BR2_PACKAGE_OPTEE_RUST_EXAMPLES_EXT_SITE
 
 if BR2_PACKAGE_OPTEE_RUST_EXAMPLES_EXT
 
+config BR2_PACKAGE_OPTEE_RUST_EXAMPLES_EXT_SDK
+	string "OPTEE SDK path"
+	default ""
+	help
+	  some help
```
Comment (jforissier) [inline br-ext/package/optee_rust_examples_ext/Config.in:18]: Please add some help. "The path to the SDK generated by the optee_rust project" for example?

**Review by jforissier — COMMENTED** [review]
LGTM with one comment below.

File: br-ext/package/optee_rust_examples_ext/Config.in (hunk 3 above)
Comment (b49020) [reply br-ext/package/optee_rust_examples_ext/Config.in:18]: Actually it rather points to `OPTEE_OS_TA_DEV_KIT_DIR` needed to build rust examples. It is defined on similar lines as `BR2_PACKAGE_OPTEE_EXAMPLES_EXT_SDK`. So how about instead "The path to OP-TEE TA development SDK"?

Comment (jforissier) [reply br-ext/package/optee_rust_examples_ext/Config.in:18]: Yes, sounds good!

**Comment by b49020** [comment]
Addressed comment, tag applied.

**Comment by b49020** [comment]
@DemesneGH can I get your ack here?

**Comment by DemesneGH** [comment]
`Acked-by: Yuan Zhuang <yuanz@apache.org>`

**Comment by b49020** [comment]
Thanks @DemesneGH, tag applied.

**Pull request merged**